public/app/features/explore/RichHistory/RichHistoryStarredTab.tsx

```diff
@@ -108,6 +112,15 @@ export function RichHistoryStarredTab(props: Props) {
               />
             </div>
           )}
+          <div className={styles.filterInput}>
+            <FilterInput
+              labelClassName="gf-form--has-input-icon gf-form--grow"
+              inputClassName="gf-form-input"
+              placeholder="Search queries"
+              value={searchFilter}
+              onChange={onSearchFilterChange}
```

Comment:
I think the onChange should be debounced so it does not kick in after each keystroke when typing rapidly. Otherwise depending on the number of items and how they are filtered this can be slow and jarring.